### ISSUE TITLE
redundantParens removes required parentheses for closures preceding nil-coalescing operators

### ISSUE THREAD

**Issue by ochristensen** [nicklockwood/SwiftFormat#389]
`
for id in a?.b.map({ $0.id }) ?? [] {
   ....
}
`
removes the parens which are required

**Comment by nicklockwood** [comment]
@ochristensen thanks for reporting, I'll push a fix asap.

**Comment by mjl5007** [comment]
It's not `redundantParens` that is removing them; it's `trailingClosures` (`--disable trailingClosures` prevents the change; `--disable redundantParens` does not).

Here's another case where `trailingClosures` is incorrectly converting to a trailing closure in a control statement:
```swift
guard case let .caseA(value) = obj.prop, !obj2!.contents.filter({ $0.id == -1 }).isEmpty else {
   return
}
```
The closure to `filter` gets converted to a trailing closure incorrectly. Curiously, it doesn't happen if you remove the `case let` condition preceding it.

**Comment by nicklockwood** [comment]
@ochristensen, @mjl5007 fixed in 0.40.2.
